### ISSUE TITLE
Be able to push Child config transformation through Baseline

### ISSUE THREAD

**Issue by MichaelNielsenDK** [umbraco/Umbraco.Cloud.Issues#22]
As of now, it is not possible to push child config transformation files through Baseline and out to the children, you would have to connect to every individual child, and push it up directly to them. 

That defies the general purpose of the baseline>child setup, where the baseline is used to control and push changes to children.  

Also, if you have to connect directly to every individual child, why then even have child transformation files. One could just push up a regular transformation file, with the content for that individual child. 

You also don't need a lot of children, before this will be a time consuming task, when you want to make changes to config files.

**Comment by sitereactor** [comment]
@MichaelNielsenDK a follow up question, are the settings always appsettings or are there types of settings you also want to be able to control via transforms?

**Comment by Frost117** [comment]
If you had blob storage setup with the children, it would be ideal if the transform would also work on the FileSystemProvider.config

A scenario would be that the baseline is connected to 1 container blob while the children of the baseline was connected to their own blob, which would be done via the transforms

**Comment by MichaelNielsenDK** [comment]
> @MichaelNielsenDK a follow up question, are the settings always appsettings or are there types of settings you also want to be able to control via transforms?

All types of config files really, i.e. Application Insights, which have it's own config file, also as @Frost117 mentions, FileSystemProvider for media blobs.

**Comment by sitereactor** [comment]
Alright, fair enough. I was mainly curious if a possible solution for this would be the ability to edit AppSettings via the Cloud Portal. But sounds like that isn't sufficient.

I'm not entirely convinced that managing transforms for a large number of child sites is a feasible solution. But will give it some more thought.

Thanks for the feedback!

**Comment by MichaelNielsenDK** [comment]
No, I'm afraid that wouldn't work. Both because, as in this example, there can be custom configuration files, and also, that option would still entail going to all children one-by-one, just as it is now, so it wouldn't really save any time or hassle.

**Comment by sitereactor** [comment]
How much of a hassle is it to handle the many configuration transforms via the Master project?
Lets say the naming convention wasn't an issue, would that make it 100% better or would it still leave you with a feeling that its "not great" or less than ideal? ... I'd imagine you probably wouldn't want ALL the transforms in each of the child sites (not sure if you already mentioned this in the other issue?).

**Comment by MichaelNielsenDK** [comment]
I'm not entirely sure if I understand you correctly, so forgive me if there's something redundant in the following :-) 

I want to be able to handle config transforms for the children via the Baseline. I would create transform files for the needed changes, push them to baseline, push them to children. 

I actually wouldn't care if the transforms are on the children, as I won't be cloning down the children, that is the point with this.

The naming convention is mostly for grouping the files together, so it's easier managed. It's a minor thing, maybe it would be even better if Cloud looked in a folder instead, but that's a whole other issue, let's keep that out of this.

Being able to just manage it from Baseline would make it a 100% better :-)

**Comment by sitereactor** [comment]
Got it, thanks! 😃 ... I saw your comment in #21 and it makes sense.

> I actually wouldn't care if the transforms are on the children, as I won't be cloning down the children, that is the point with this.

That is fair enough, but as the amount of child projects grows I'd imagine it would become annoying in certain scenarios. Given something like a "system transform folder" we could probably do something clever with naming conventions and how its applied to the children, so the many transforms would be taken out of the git repository and thus not be deployed to all the child projects.

Thanks again for your input and feedback. It's very appreciated!

**Comment by lukehook** [comment]
> That is fair enough, but as the amount of child projects grows I'd imagine it would become annoying in certain scenarios. Given something like a "system transform folder" we could probably do something clever with naming conventions and how its applied to the children, so the many transforms would be taken out of the git repository and thus not be deployed to all the child projects.

In my opinion, having all the child transforms pushed through the Baseline project is definitely a better solution than what is available currently. Cloning down and pushing a transform on a per-child basis is not really sustainable and does negate what is a HUGE benefit of the baselines concept. However, I completely understand what you're saying, if there were say, 1000 children and they all had a transform there is going to be a whole lot of redundant transform files within the root (or wherever) of EVERY child. Naming conventions could work to make this a much cleaner approach as you say.

If it could 'look up' the specific transform and apply it based on name that would be great.

**Comment by nul800sebastiaan** [comment]
Just as an update, we've not been able to prioritize this feature request and are putting it on our list of future ideas to work on around baselines when we get a bunch of baselines features that we can work on in a bundle.
